src/api/github/checkRun.ts

```diff
@@ -0,0 +1,120 @@
+import { Octokit } from '@octokit/rest';
+import { Endpoints } from '@octokit/types';
+
+type Output = NonNullable<
+  Endpoints['POST /repos/{owner}/{repo}/check-runs']['parameters']['output']
+>;
+
+export type Annotation = NonNullable<Output['annotations']>[number];
+
+const GITHUB_MAX_ANNOTATIONS = 50;
+
+const isGitHubAnnotationsEnabled = (): boolean =>
+  Boolean(
+    process.env.BUILDKITE_BUILD_NUMBER &&
+      process.env.BUILDKITE_COMMIT &&
+      process.env.BUILDKITE_REPO &&
+      process.env.GITHUB_API_TOKEN,
+  );
+
+// Pulls out the GitHub Owner + Repo String from repo urls eg.
+// git@github.com:seek-oss/skuba.git
+// https://github.com/seek-oss/skuba.git
+// Pulls out `seek-oss` as owner and `skuba` as repo
+const ownerRepoRegex = new RegExp(/github.com(?::|\/)(.*)\/(.*).git/);
+
+const getOwnerRepo = (): { owner: string; repo: string } => {
+  const match = ownerRepoRegex.exec(process.env.BUILDKITE_REPO as string);
+  const owner = match?.[1];
+  const repo = match?.[2];
+
+  if (!owner || !repo) {
+    throw new Error(
+      'Could not extract GitHub owner/repo from BUILDKITE_REPO environment variable',
+    );
+  }
+
+  return {
+    owner,
+    repo,
+  };
+};
+
+/**
+ * Create a uniform title format for our check runs
+ * @param conclusion - `failure` or `success`
+ * @param annotationsLength - Number of annotations added
+ * @returns Title eg. Build #12 failed (24 annotations added)
```

Comment:
```suggestion
 * Create a uniform title format for our check runs, e.g.
 *
 * ```text
 * Build #12 failed (24 annotations added)
 * ```
```